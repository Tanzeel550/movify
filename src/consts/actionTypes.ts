export type EmailPassType = {
  email: string;
  password: string;
};

export type AuthActionsType = {
  type: 'LOGIN' | 'LOGOUT';
  user?: object;
};

export type AuthUserType = {
  user: object;
};

export type MoviesActionReturnType = {
  type: 'CREATE_MOVIE' | 'GET_MOVIES' | 'UPDATE_MOVIE' | 'DELETE_MOVIE';
  movie?: object;
  movies?: object;
  id?: string;
};

export type MoviesActionParamsType = {
  id?: string;
  movie?: object;
  movies?: object;
};

export type SEARCH_TYPE = {
  text: string;
};

export type ErrorType = {
  message: string;
  title: string;
};

export type ErrorActionsReturnType = {
  type: 'SET_ERROR' | 'CLEAR_ERROR';
  message?: string;
  title?: string;
};
