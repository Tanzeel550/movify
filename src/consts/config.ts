const OMBD_API_KEY = process.env.REACT_APP_OMDB_API_KEY;
export const SEARCH_MOVIE_BY_TEXT = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&s=`;
export const GET_MOVIE_BY_TITLE = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&t=`;
export const GET_MOVIE_BY_IMDB_ID = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&i=`;

export const PROMISE_AREAS = {
  SEARCH_MOVIE_BY_TEXT: 'SEARCH_MOVIE_BY_TEXT',
  GET_MOVIE_BY_TITLE: 'GET_MOVIE_BY_TITLE',
  GET_ALL_MOVIES: 'GET_ALL_MOVIES',
  LOAD_PAGE: 'LOAD_PAGE',
};

export const skills1 = [
  'Web Design',
  'JavaScript Developer',
  'React JS',
  'Rest API',
  'NodeJS',
];

export const skills2 = ['Company', 'Team', 'Careers'];

export const TEST_MOVIES_SRC = 'movies_test';
