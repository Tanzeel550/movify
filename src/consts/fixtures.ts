import {
  APIMovieResultByTitle,
  APISingleMovieResultBySearch,
  FireDBMovieItem,
} from '../types/APITypes';
import { AuthUserType } from '../types/AuthTypes';

export const movies: FireDBMovieItem[] = [
  {
    id: '1234567',
    name: 'The Dark Knight',
    watched: true,
    dateWatched: '234567890',
    whatYouLearnt: 'The world is as good as we let it to be',
    Poster: 'https://image.com',
    createdAt: '9876543',
  },
  {
    id: '4567890',
    name: 'GodFather',
    watched: false,
    dateWatched: '98765434567',
    whatYouLearnt: 'You can act like a man what is the matter with you?',
    Poster: 'https://godfather.com',
    createdAt: '987654567890',
  },
  {
    id: '9ikjnbvfrty',
    name: 'Avengers: EndGame',
    watched: false,
    dateWatched: '',
    whatYouLearnt: 'Thanos will be defeated',
    Poster: 'https://avengers.com',
    createdAt: '69864123340404',
  },
];

export const myUser: AuthUserType = {
  user: {
    name: 'Tanzeel',
    email: 'tanzeelahmed550@gmail.com',
    passion: 'Web Dev',
    os: 'Kali Linux',
    version: '1.0.0',
    height: 5.5,
  },
};

export const skills1 = [
  'Web Design',
  'JavaScript Developer',
  'React JS',
  'Rest API',
  'NodeJS',
];

export const skills2 = ['Company', 'Team', 'Careers'];

export const email = 'unknownPerson123@uet.com';
export const password = 'JonSnow';
export const nameOfTheMovie = 'Godfather';

export const movieFromAPIByTitle: APIMovieResultByTitle = {
  Title: 'The Dark Knight',
  Year: '2008',
  Rated: 'PG-13',
  Released: '18 Jul 2008',
  Runtime: '152 min',
  Genre: 'Action, Crime, Drama, Thriller',
  Director: 'Christopher Nolan',
  Writer:
    'Jonathan Nolan (screenplay), Christopher Nolan (screenplay), Christopher Nolan (story), David S. Goyer (story), Bob Kane (characters)',
  Actors: 'Christian Bale, Heath Ledger, Aaron Eckhart, Michael Caine',
  Plot:
    'When the menace known as the Joker wreaks havoc and chaos on the people of Gotham, Batman must accept one of the greatest psychological and physical tests of his ability to fight injustice.',
  Language: 'English, Mandarin',
  Country: 'USA, UK',
  Awards: 'Won 2 Oscars. Another 157 wins & 163 nominations.',
  Poster:
    'https://m.media-amazon.com/images/M/MV5BMTMxNTMwODM0NF5BMl5BanBnXkFtZTcwODAyMTk2Mw@@._V1_SX300.jpg',
  Ratings: [
    { Source: 'Internet Movie Database', Value: '9.0/10' },
    {
      Source: 'Rotten Tomatoes',
      Value: '94%',
    },
    { Source: 'Metacritic', Value: '84/100' },
  ],
  Metascore: '84',
  imdbRating: '9.0',
  imdbVotes: '2,341,244',
  imdbID: 'tt0468569',
  Type: 'movie',
  DVD: '14 Jun 2010',
  BoxOffice: '$534,858,444',
  Production: 'Syncopy',
  Website: 'N/A',
  Response: 'True',
};

export const moviesFromAPIBySearch: APISingleMovieResultBySearch[] = [
  {
    Title: 'The Dark Knight',
    Year: '2008',
    imdbID: 'tt0468569',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMTMxNTMwODM0NF5BMl5BanBnXkFtZTcwODAyMTk2Mw@@._V1_SX300.jpg',
  },
  {
    Title: 'The Dark Knight Rises',
    Year: '2012',
    imdbID: 'tt1345836',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMTk4ODQzNDY3Ml5BMl5BanBnXkFtZTcwODA0NTM4Nw@@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Dark Knight Returns, Part 1',
    Year: '2012',
    imdbID: 'tt2313197',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMzIxMDkxNDM2M15BMl5BanBnXkFtZTcwMDA5ODY1OQ@@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Dark Knight Returns, Part 2',
    Year: '2013',
    imdbID: 'tt2166834',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BYTEzMmE0ZDYtYWNmYi00ZWM4LWJjOTUtYTE0ZmQyYWM3ZjA0XkEyXkFqcGdeQXVyNTA4NzY1MzY@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Dark Knight Returns',
    Year: '2013',
    imdbID: 'tt11060882',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMjExMGE4NDQtM2MwMS00YzhlLTlkYjgtNmExNzM1ZDBhYzg1XkEyXkFqcGdeQXVyMTY5Nzc4MDY@._V1_SX300.jpg',
  },
];
