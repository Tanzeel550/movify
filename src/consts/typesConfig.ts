export type EmailPassType = {
  email: string;
  password: string;
};

export type AuthActionsType = {
  type: 'LOGIN' | 'LOGOUT';
  user?: object;
};

export type AuthUserType = {
  user: object;
};

export type MOVIES_ACTION = {
  type: 'CREATE_MOVIE' | 'GET_MOVIES' | 'UPDATE_MOVIE' | 'DELETE_MOVIE';
  movie?: object;
  movies?: object;
  id?: string;
};

export type SEARCH_TYPE = {
  text: string;
};

export type MOVIE_ID = {
  id?: string;
  movie?: object;
  movies?: object;
};
