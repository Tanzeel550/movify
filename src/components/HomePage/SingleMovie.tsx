import { NavLink } from 'react-router-dom';
import React from 'react';
import { FireDBMovieItem } from '../../consts/actionTypes';

type Props = {
  movie: FireDBMovieItem;
};

const SingleMovie = ({ movie }: Props) => (
  <NavLink className="movie__item" to={`/update/${movie.id}`}>
    <div className="movie__item--poster">
      <img src={movie.Poster} alt="" />
    </div>

    <div className="movie__item--text">
      <h5 className="movie__item--text-text">{movie.name}</h5>
      {movie.watched ? (
        <div>
          <p>You watched this movie on {movie.dateWatched}</p>
          <h5>What You Learned from it?</h5>
          <p className=".u-tex-ellipsis">
            {movie.whatYouLearnt || 'Nothing Learned... Just watched! '}
          </p>
        </div>
      ) : (
        'You really need to watch this movie'
      )}
    </div>
  </NavLink>
);

export default SingleMovie;