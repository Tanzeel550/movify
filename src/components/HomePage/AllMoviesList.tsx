import React from 'react';
import { FireDBMovieItem } from '../../consts/actionTypes';
import SingleMovie from './SingleMovie';

type Props = {
  movies: FireDBMovieItem[];
};

const AllMoviesList: React.FC<Props> = ({ movies }: Props) => (
  <div className="all__movies">
    {movies.map((movie, i) => (
      <SingleMovie movie={movie} key={i} />
    ))}
  </div>
);

export default AllMoviesList;
