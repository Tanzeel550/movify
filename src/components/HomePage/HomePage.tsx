import React from 'react';
import { connect, ConnectedProps } from 'react-redux';
import AllMoviesList from './AllMoviesList';
import { PROMISE_AREAS } from '../../consts/config';
import PromiseLoading from '../Utils/PromiseLoading';
import { FireDBMovieItem } from '../../consts/actionTypes';

const getFirstName = (name: string) => {
  const firstName = name.split(
    /[`1234567890!@#£€$¢¥§%°^&*()-_+={}[]|\/:;"'<>,.?]+/
  )[0];
  return firstName[0].toUpperCase() + firstName.substring(1);
};

type StateTypes = {
  movies: FireDBMovieItem[];
  auth: {
    user: {
      name: string;
      email: string;
    };
  };
};

const mapStateToProps = (state: StateTypes) => ({
  movies: state.movies,
  name: state.auth.user.name || state.auth.user.email,
});

const connector = connect(mapStateToProps);
type PropsFromRedux = ConnectedProps<typeof connector>;

type Props = PropsFromRedux;

const HomePage: React.FC<Props> = ({ movies, name }: Props) => {
  return (
    <div>
      <div className="home__user--info">
        <div className="container u-text-white">
          <span>Hi </span>
          <span className="u-text-large">{getFirstName(name)}</span>,
          <p className="u-text-white">Welcome to Movify 😀️</p>
          <p className="u-text-white">Add Movies and make them YOURS!</p>
        </div>
      </div>

      <div className="container">
        <PromiseLoading area={PROMISE_AREAS.GET_ALL_MOVIES} />
        {movies.length > 0 ? (
          <AllMoviesList movies={movies} />
        ) : (
          <h1 className="u-text-colorized">Please add Movies to Continue...</h1>
        )}
      </div>
    </div>
  );
};

export default connector(HomePage);
