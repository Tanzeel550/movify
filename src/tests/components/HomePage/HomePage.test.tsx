import { shallow } from 'enzyme';
import { HomePage } from '../../../components/HomePage/HomePage';
import { email, movies } from '../../../consts/fixtures';
import React from 'react';

test('HomePage should be rendered correctly', () => {
  // @ts-ignore
  const wrapper = shallow(<HomePage movies={movies} name="Mike Ahmed" />);
  expect(wrapper).toMatchSnapshot();
});

test('Home Page should not display any movie', () => {
  // @ts-ignore
  const wrapper = shallow(<HomePage movies={[]} name={email} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('h1').text()).toMatch('Please add Movies to Continue...');
});

export {};
