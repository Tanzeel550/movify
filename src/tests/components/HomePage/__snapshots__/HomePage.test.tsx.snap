// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should not display any movie 1`] = `
<div>
  <div
    className="home__user--info"
  >
    <div
      className="container u-text-white"
    >
      <span>
        Hi 
      </span>
      <span
        className="u-text-large"
      >
        Unknown
      </span>
      ,
      <p
        className="u-text-white"
      >
        Welcome to Movify üòÄÔ∏è
      </p>
      <p
        className="u-text-white"
      >
        Add Movies and make them YOURS!
      </p>
    </div>
  </div>
  <div
    className="container"
  >
    <h1
      className="u-text-colorized"
    >
      Please add Movies to Continue...
    </h1>
  </div>
</div>
`;

exports[`HomePage should be rendered correctly 1`] = `
<div>
  <div
    className="home__user--info"
  >
    <div
      className="container u-text-white"
    >
      <span>
        Hi 
      </span>
      <span
        className="u-text-large"
      />
      ,
      <p
        className="u-text-white"
      >
        Welcome to Movify üòÄÔ∏è
      </p>
      <p
        className="u-text-white"
      >
        Add Movies and make them YOURS!
      </p>
    </div>
  </div>
  <div
    className="container"
  >
    <AllMoviesList
      movies={
        Array [
          Object {
            "Poster": "https://image.com",
            "createdAt": "9876543",
            "dateWatched": "234567890",
            "id": "1234567",
            "name": "The Dark Knight",
            "updatedAt": "",
            "watched": true,
            "whatYouLearnt": "The world is as good as we let it to be",
          },
          Object {
            "Poster": "https://godfather.com",
            "createdAt": "987654567890",
            "dateWatched": "98765434567",
            "id": "4567890",
            "name": "GodFather",
            "updatedAt": "",
            "watched": false,
            "whatYouLearnt": "You can act like a man what is the matter with you?",
          },
          Object {
            "Poster": "https://avengers.com",
            "createdAt": "69864123340404",
            "dateWatched": "",
            "id": "9ikjnbvfrty",
            "name": "Avengers: EndGame",
            "updatedAt": "",
            "watched": false,
            "whatYouLearnt": "Thanos will be defeated",
          },
        ]
      }
    />
  </div>
</div>
`;
